app/views/pages/use-cases/using-rest-endpoints-with-react.liquid

```diff
@@ -0,0 +1,300 @@
+%}
+
+{{ result }}
+```
+
+As you can see, you can define your endpoint's slug and method, you don't need a layout and you can use your previously created `headers.liquid` file to render you response header.
+
+In the logic part, you can use your previously created graphql file with the parameters from the context and based on that's result you can set an error status code. As the result is in `json` format, just render it and it will contain everything.
+
+### Step 7: Use your REST endpoint in your React app
+
+Now you have everything to store a user so you can modify your React app's `/my-app/src/App.tsx` and append a form on it. In this example you will use typescript and react hook form to implement the POST request and you can see the created user's ID in the JavaScript console.
+
+We _strongly recommend_ to use [axios](https://github.com/axios/axios) as the `csrf-token` is very important and it's easy to set up with:
+
+```javascript
+axios.defaults.xsrfCookieName = 'CSRF-TOKEN';
+axios.defaults.xsrfHeaderName = 'X-CSRF-Token';
+```
+
+So install and use that instead of `fetch`
+
+```shell
+/my-app$ npm install axios --save
+```
+
+```tsx
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
+
+import logo from './logo.svg';
+import './App.css';
+
+axios.defaults.xsrfCookieName = 'CSRF-TOKEN';
+axios.defaults.xsrfHeaderName = 'X-CSRF-Token';
+
+type Inputs = {
+  name: string;
+  email: string;
+};
+
+function App() {
+  const { register, handleSubmit, errors } = useForm<Inputs>();
+
+  async function onSubmit(data: Inputs) {
+    try {
+      const response = await axios.post('/api/users', data);
+      console.log(response);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <p>Create a new user</p>
+
+        <div>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <input
+              name="name"
+              placeholder="Name"
+              ref={register({
+                required: 'Required',
+              })}
+            />
+            {errors.name && errors.name.message}
+
+            <input
+              name="email"
+              placeholder="Email"
+              ref={register({
+                required: 'Required',
+                pattern: {
+                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+                  message: 'invalid email address',
+                },
+              })}
+            />
+            {errors.email && errors.email.message}
+
+            <button type="submit">Save</button>
+          </form>
+        </div>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+### Step 8: Set up proxy for local API calls
+
+Due to security issues you should not use:
+
+```
+response_headers: >
+  {
+    "Access-Control-Allow-Origin": "*"
+  }
+```
```

Comment:
This whole section tells the user what NOT to do. We don’t really have to add this information here, especially that it is actually not enough to make the API open to external calls. This could be a subject of another article altogether.

```suggestion
```